wordpress-mobile/WordPress-iOS
# Stats Revamp: Views Visitors Card and Graph

This PR adds a new Views / Visitors Graph and Card to the Insights screen as part of Stats Revamp

Fixes #18234
Fixes #18235

Note that this PR
 * does not handle Ghosting / Ghost card yet
 * does not handle the containsCacheData call correctly.  Currently ViewModels are 1:1 with Stores
These will be taken care of in separate PRs

Design
<img width="207" alt="image" src="https://user-images.githubusercontent.com/88816724/166713203-d8f3a40d-a055-448f-8aa8-8b538cc1b7ca.png">

Implementation
<img width="220" alt="image" src="https://user-images.githubusercontent.com/88816724/166713488-489f0cd5-4cab-40a8-82f6-a642f44120b4.png">



To test:
1. First, build and run, navigate to Stats Insights for a site and ensure that there is no option to add a Views & Visitors card like in s/shot below
<img width="221" alt="image" src="https://user-images.githubusercontent.com/88816724/166713693-f8e3a419-dd1b-44db-8a35-62b5d4384c61.png">

2. Now enable both of the new stats feature flags and build and run again
	
```
	statsNewAppearance
	statsNewInsights
```
	
3. Add New Stats Card = Views & Visitors
	
4. New Views & Visitors Card and Graph should be added to the screen like the Implementation screenshot above

5. Tap on the Graph.  
		a. Tapping on the Graph will select the nearest whole day to where you have tapped
		b. A marker should appear with the information that was also in the design
		c. A dotted vertical line should appear on the day that was displayed
		d. A circle / dot will display on the Y axis for the selected date at the Y intercept
6. Tap on the edges of the Graph
		a. The stem of the marker should change to nicely render the marker away from the edges like in the following screenshot(s)
<img width="210" alt="image" src="https://user-images.githubusercontent.com/88816724/166713746-bc2ec3a6-6e08-4c7b-8a55-4371e7d4c7ab.png">
<img width="207" alt="image" src="https://user-images.githubusercontent.com/88816724/166713789-14c42cda-edb9-4736-b01d-a98a0deb5324.png">


## Regression Notes
1. Potential unintended areas of impact
Stats Insights

2. What I did to test those areas of impact (or what existing automated tests I relied on)
Manually tested + created a UnitTest for Domain object usage in ViewModel

3. What automated tests I added (or what prevented me from doing so)
ViewModel UnitTest

PR submission checklist:
